python_modules/dagster/dagster/_core/definitions/base_asset_graph.py

```diff
@@ -267,6 +267,10 @@ def root_executable_asset_keys(self) -> AbstractSet[AssetKey]:
     @abstractmethod
     def asset_check_keys(self) -> AbstractSet[AssetCheckKey]: ...
 
+    @cached_property
+    def orphan_asset_check_keys(self) -> AbstractSet[AssetCheckKey]:
```

Comment:
I think we need to define 'orphan'- checks on non observable source assets and assets in other code locations?